### PR TITLE
Updates userControllers status code for Update and Delete

### PR DESCRIPTION
**Pull request by mattfeg**
Updates userControllers status code for Update and Delete.
Updates userControllers tests for changes.

### CODE REVIEW

File: coverage/.last_run.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "line": 84.21
+    "line": 80.95
```
Comment (edimossilva) [inline coverage/.last_run.json:3]: Add this file to gitignore

File: coverage/.resultset.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "RSpec": {
     "coverage": {
-      "/home/ferreira/Repositories/rails-01/app/controllers/users_controller.rb": {
+      "/Users/ferreira/Repositories/rails-01/app/controllers/users_controller.rb": {
```
Comment (edimossilva) [inline coverage/.resultset.json:4]: Add this file to gitignore

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -20,19 +20,27 @@ def create
 
   def update
     @user = User.find(params[:id])
-    if @user.update(user_params)
-      render json: @user
+    if @user
```
Comment (edimossilva) [inline app/controllers/users_controller.rb:23]: the code for not_found is being duplicated at these 2 methods, please refactor in a way that you will not need to duplicate it

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -20,19 +20,27 @@ def create
...
     end
   end
 
   def destroy
     @user = User.find(params[:id])
-    if @user.destroy
-      render json: @user
+    if @user
+      if @user.destroy
+        render json: @user, status: :no_content
```
Comment (edimossilva) [inline app/controllers/users_controller.rb:38]: you can return an empty hash instead of a user here

**Review by edimossilva — CHANGES_REQUESTED** [review]
Add specs to cover every single change you made

File: coverage/.last_run.json (hunk 1 above)
Comment (mattfeg) [reply coverage/.last_run.json:3]: Resolved on PR #6

File: coverage/.resultset.json (hunk 2 above)
Comment (mattfeg) [reply coverage/.resultset.json:4]: Resolved on PR #6

**Review by edimossilva — CHANGES_REQUESTED** [review]
Add specs to not found for update and destroy methods

File: spec/controllers/user_controller_spec.rb [hunk 5]
```diff
@@ -126,15 +126,22 @@
 
     it "returns a sucessful response" do
       delete :destroy, params: { id: user.id }
-      expect(response).to have_http_status(:ok)
+      expect(response).to have_http_status(:no_content)
     end
 
     it "returns a unprocessable entity response" do
-      allow(User).to receive(:find).and_return(user)
+      allow(User).to receive(:find_by).and_return(user)
```
Comment (edimossilva) [inline spec/controllers/user_controller_spec.rb:133]: remove all mocks from these specs, mocks should be used for complex stuff

**Pull request merged**